File: leetcode/0026. 删除排序数组中的重复项/go/remove-duplicates-from-sorted-array.go
package problem0026

func removeDuplicates(a []int) int {
	p, r, size := 0, 1, len(a)
	for ; r < size; r++ {
		if !(a[r]==a[p]) {
			p++
			a[p], a[r] = a[r], a[p]
		}
	}
	return p+1
}

File: leetcode/0026. 删除排序数组中的重复项/python/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        p = 0
        for r in range(1, len(nums)):
            if nums[p] != nums[r]:
                p += 1
                nums[p], nums[r] = nums[r], nums[p]
        return p+1
